core/lib/object_store/src/objects.rs

```diff
@@ -63,6 +67,62 @@ macro_rules! serialize_using_bincode {
     };
 }
 
+/// Derives [`StoredObject::serialize()`] and [`StoredObject::deserialize()`] using
+/// the `json` (de)serializer. Should be used in `impl StoredObject` blocks.
+
+impl StoredObject for SnapshotFactoryDependencies {
+    const BUCKET: Bucket = Bucket::StorageSnapshot;
+    type Key<'a> = L1BatchNumber;
+
+    fn encode_key(key: Self::Key<'_>) -> String {
+        format!("snapshot_l1_batch_{key}_factory_deps.json.gzip")
+    }
+
+    //TODO use better language agnostic serialization format like protobuf
+    fn serialize(&self) -> Result<Vec<u8>, BoxedError> {
+        let mut encoder = GzEncoder::new(Vec::new(), Compression::default());
+        serde_json::to_writer(&mut encoder, self).map_err(BoxedError::from)?;
+        encoder.finish().map_err(From::from)
+    }
+
+    fn deserialize(bytes: Vec<u8>) -> Result<Self, BoxedError> {
+        let mut decoder = GzDecoder::new(&bytes[..]);
+        let mut decompressed_bytes = Vec::new();
+        decoder
+            .read_to_end(&mut decompressed_bytes)
+            .map_err(BoxedError::from)?;
+        serde_json::from_slice(&decompressed_bytes).map_err(From::from)
+    }
+}
+
+impl StoredObject for SnapshotStorageLogsChunk {
+    const BUCKET: Bucket = Bucket::StorageSnapshot;
+    type Key<'a> = SnapshotStorageLogsStorageKey;
+
+    fn encode_key(key: Self::Key<'_>) -> String {
+        format!(
+            "snapshot_l1_batch_{}_storage_logs_part_{:0<3}.json.gzip",
```

Comment:
Shouldn't it be `{:0>3}` (i.e., the formatter chunk is *right*-aligned)? Could you add a unit test to cover this?

Reply:
it definitely should! I will add a test for that